### PR TITLE
d_a_guard_mng OK, d_a_obj_smoke OK, d_a_tag_smk_emt OK, d_a_hitobj OK

### PR DESCRIPTION
**Pull request by Pheenoh**

### CODE REVIEW

File: include/rel/d/a/d_a_hitobj/d_a_hitobj.h [hunk 1]
```diff
@@ -2,5 +2,20 @@
...
+#include "d/com/d_com_inf_game.h"
+
+class hitobj_class : public fopAc_ac_c {
+public:
+    /* 0x568 */ request_of_phase_process_class mPhase;
+    /* 0x570 */ u8 field_0x570;
+    /* 0x571 */ u8 field_0x571;
+    /* 0x572 */ s16 field_0x572;
+    /* 0x574 */ dCcD_Stts field_0x574;
+    /* 0x5B0 */ dCcD_SrcSph field_0x5b0; // probably wrong
```
Comment (TakaRikka) [inline include/rel/d/a/d_a_hitobj/d_a_hitobj.h:15]: this looks like it should be `dCcD_Sph`, with the below members removed

File: rel/d/a/d_a_guard_mng/d_a_guard_mng.cpp [hunk 2]
```diff
@@ -1,171 +1,108 @@
...
 
 /* 80837C6C-80837D44 0001CC 00D8+00 1/0 0/0 0/0 .text daGuardMng_Execute__FP12daGuardMng_c */
-#pragma push
-#pragma optimization_level 0
-#pragma optimizewithasm off
-static asm void daGuardMng_Execute(daGuardMng_c* param_0) {
-    nofralloc
-#include "asm/rel/d/a/d_a_guard_mng/d_a_guard_mng/daGuardMng_Execute__FP12daGuardMng_c.s"
+static int daGuardMng_Execute(daGuardMng_c* i_this) {
+    return i_this->execute(data_80837DE4[0]); // this inline technically doesn't take a param. small hack to get the match
```
Comment (TakaRikka) [inline rel/d/a/d_a_guard_mng/d_a_guard_mng.cpp:57]: whats the specific issue here? definitely looks very sus

File: rel/d/a/d_a_hitobj/d_a_hitobj.cpp [hunk 3]
```diff
@@ -82,63 +36,51 @@ extern "C" void Set__8dCcD_SphFRC11dCcD_SrcSph();
...
-#pragma optimization_level 0
-#pragma optimizewithasm off
-static asm void daHitobj_Execute(hitobj_class* param_0) {
-    nofralloc
-#include "asm/rel/d/a/d_a_hitobj/d_a_hitobj/daHitobj_Execute__FP12hitobj_class.s"
+static int daHitobj_Execute(hitobj_class* i_this) {
+    if (i_this->field_0x572 != 0) {
+        i_this->field_0x572--;
+        i_this->field_0x6d4.SetC(i_this->current.pos);
+        dComIfG_Ccsp()->Set((cCcD_Obj*)&i_this->field_0x5b0);
```
Comment (TakaRikka) [inline rel/d/a/d_a_hitobj/d_a_hitobj.cpp:60]: this cast should be able to be removed once the member is fixed to `dCcD_Sph`

File: rel/d/a/d_a_hitobj/d_a_hitobj.cpp [hunk 4]
```diff
@@ -82,63 +36,51 @@ extern "C" void Set__8dCcD_SphFRC11dCcD_SrcSph();
...
+static int daHitobj_Delete(hitobj_class* i_this) {
+    dComIfG_resDelete(&i_this->mPhase,"Hitobj");
+    return 1;
 }
-#pragma pop
 
 /* ############################################################################################## */
 /* 8046B5DC-8046B61C 000000 0040+00 1/1 0/0 0/0 .data            cc_sph_src$3657 */
+// 
 SECTION_DATA static u8 cc_sph_src[64] = {
```
Comment (TakaRikka) [inline rel/d/a/d_a_hitobj/d_a_hitobj.cpp:84]: this should be a local static dCcD_SrcSph if it doesn't cause other problems

File: rel/d/a/d_a_guard_mng/d_a_guard_mng.cpp (hunk 2 above)
Comment (Pheenoh) [reply rel/d/a/d_a_guard_mng/d_a_guard_mng.cpp:57]: static data is in the TU, but used in the header via the inline so the inline can't find it. resolved by just moving the static data to the header

File: rel/d/a/d_a_hitobj/d_a_hitobj.cpp (hunk 4 above)
Comment (Pheenoh) [reply rel/d/a/d_a_hitobj/d_a_hitobj.cpp:84]: For some reason the .data section is too small even after setting this up

**Comment by Pheenoh** [comment]
All issues fixed except the last one, not sure how to setup `cc_sph_src` correctly?

**Review by TakaRikka — APPROVED** [review]

**Pull request merged**